### PR TITLE
[lexical-markdown] Bug Fix: run element markdown shortcuts on Enter

### LINKED ISSUE #7358: Bug: Can't trigger markdown shortcuts for code blocks when pressing enter
Hello, I've overridden the CODE transformer to update the startRegexp to try to trigger the code block as soon as we type 3 backticks + enter or space (which feels more natural than 3 backticks + space)

However the code here seems to **always** expect a space at the end to trigger the shortcut (particularly `if (textContent[anchorOffset - 1] !== ' ') { `)
https://github.com/facebook/lexical/blob/39be11428b5a35a6c3625babdc2514f5bc8ad4bc/packages/lexical-markdown/src/MarkdownShortcuts.ts#L57-L68

Lexical version: any version

## Steps To Reproduce

1. Override the CODE transformer with a start regexp that includes carriage return e.g. 
```
export const CUSTOM_TRANSFORMERS: Array<Transformer> = [
  {
    ...CODE,
    regExpStart: /^[ \t]*```(\w+)?[\r\n\s]/,
  },
]
```
2. Notice that the code block never triggers on enter, due to the fixed checks for space

## The current behavior
Any markdown shortcut regexp does not support carriage return

## The expected behavior
Carriage return should be supported, perhaps with a flag, if the current behaviour is necessary for most of the shortucts.

## Impact of fix
More freedom on markdown shortcuts

### PR DESCRIPTION
**Pull request by abhinavgautam01**
Element markdown shortcuts required a space before the caret and were not
invoked from KEY_ENTER_COMMAND, so patterns ending at EOL (or pasted lines)
never fired. Align element behavior with multiline Enter handling and add tests.

## Description

**Current behavior:** Block-level **element** markdown transformers (e.g. `HEADING`, `QUOTE`, list prefixes—or a custom `element` transformer such as a code-fence pattern) only ran from the markdown shortcut update path, which insisted on a space immediately before the caret. They were not run from `KEY_ENTER_COMMAND`, so shortcuts that naturally end at the end of the line (or text inserted in one shot so that space-based path never ran) did not transform.
**This PR:** Extends `runElementTransformers` with a `triggerOnEnter` mode that skips the forced space check and matches the markdown prefix up to the caret when it sits at end-of-text, and invokes that from `KEY_ENTER_COMMAND` after multiline transformers (same EOL caret guard as today). Adds unit tests for a custom element fence (` ``` ` + Enter) and for `## ` pasted/inserted at once + Enter.
Closes #7358
### Before
- `pnpm run test-unit -- packages/lexical-markdown/src/__tests__/unit/LexicalMarkdown.test.ts` (existing suite only; Enter + element edge cases not covered).
### After
- `pnpm run test-unit -- packages/lexical-markdown/src/__tests__/unit/LexicalMarkdown.test.ts` — all passing, including new tests:
  - custom element transformer: ` ``` ` then Enter → empty code block
  - `HEADING`: insert `## ` in one update then Enter → `<h2>`

### CODE REVIEW

File: packages/lexical-markdown/src/MarkdownShortcuts.ts [hunk 1]
```diff
@@ -622,6 +627,21 @@ export function registerMarkdownShortcuts(
...
+            anchorOffset,
+            byType.element,
+            true,
+          )
+        ) {
+          if (event !== null) {
+            event.preventDefault();
+          }
+          return true;
+        }
```
Comment (etrepum) [inline packages/lexical-markdown/src/MarkdownShortcuts.ts]: If you combine the if statements with an `||` then you don't need the redundant code to return and preventDefault

Comment (abhinavgautam01) [reply packages/lexical-markdown/src/MarkdownShortcuts.ts]: done, merged the two branches with runMultilineElementTransformers(...) || runElementTransformers(...) so we only have one preventDefault() + return true path...

**Review by etrepum — COMMENTED** [review]
In order to ensure that it doesn't cause misbehavior and to be able to quickly QA the feature we should enable these in the playground transformers where the triggerOnEnter behavior makes sense. Right now the potential problems with this code are hidden because the only time they're exercised is in one unit test.

**Comment by abhinavgautam01** [comment]
> In order to ensure that it doesn't cause misbehavior and to be able to quickly QA the feature we should enable these in the playground transformers where the triggerOnEnter behavior makes sense. Right now the potential problems with this code are hidden because the only time they're exercised is in one unit test.

Added optional triggerOnEnter on element / multiline element transformers and wired KEY_ENTER_COMMAND to only those transformers. Core TRANSFORMERS set triggerOnEnter: true on the block shortcuts that already relied on Enter in tests (headings, quote, lists, code fence). Playground enables it on HR and TWEET as well; MarkdownShortcutPlugin’s HR matches. TABLE is omitted for now to avoid fragile Enter behavior on incomplete rows.

File: packages/lexical-markdown/src/MarkdownShortcuts.ts [hunk 2]
```diff
@@ -436,6 +441,12 @@ export function registerMarkdownShortcuts(
   transformers: Array<Transformer> = TRANSFORMERS,
 ): () => void {
   const byType = transformersByType(transformers);
+  const elementTransformersForEnter = byType.element.filter(
+    t => t.triggerOnEnter === true,
+  );
+  const multilineElementTransformersForEnter = byType.multilineElement.filter(
+    t => t.triggerOnEnter === true,
+  );
```
Comment (etrepum) [inline packages/lexical-markdown/src/MarkdownShortcuts.ts]:
```suggestion
  const elementTransformersForEnter = byType.element.filter(
    t => t.triggerOnEnter,
  );
  const multilineElementTransformersForEnter = byType.multilineElement.filter(
    t => t.triggerOnEnter,
  );
```

The `=== true` is redundant, it's either boolean or undefined which is equivalent to false

Comment (abhinavgautam01) [reply packages/lexical-markdown/src/MarkdownShortcuts.ts]: food call, triggerOnEnter is optional and we only care whether it’s truthy, so I dropped the explicit === true and kept both filters as t => t.triggerOnEnter. change is pushed...

**Comment by abhinavgautam01** [comment]
ping @etrepum

File: packages/lexical-markdown/src/MarkdownShortcuts.ts [hunk 3]
```diff
@@ -61,18 +62,22 @@ function runElementTransformers(
...
-    if (
-      match &&
-      match[0].length ===
-        (match[0].endsWith(' ') ? anchorOffset : anchorOffset - 1)
-    ) {
+    const expectedMatchLength = triggerOnEnter
+      ? anchorOffset
+      : match && match[0].endsWith(' ')
+        ? anchorOffset
+        : anchorOffset - 1;
```
Comment (etrepum) [inline packages/lexical-markdown/src/MarkdownShortcuts.ts]:
```suggestion
    const expectedMatchLength = (triggerOnEnter || match && match[0].endsWith(' '))
        ? anchorOffset
        : anchorOffset - 1;
```

These can be fused

File: packages/lexical-markdown/src/MarkdownTransformers.ts [hunk 4]
```diff
@@ -680,6 +693,8 @@ export const CODE: MultilineElementTransformer = {
       })(rootNode, children, startMatch, isImport);
     }
   },
+
+  triggerOnEnter: true,
```
Comment (etrepum) [inline packages/lexical-markdown/src/MarkdownTransformers.ts]: The playground handles the triple-backtick trigger on enter now without this code, so I think either this is not necessary for multiline-element or there's more code that can be cleaned up as a result of this change.

File: packages/lexical-markdown/src/MarkdownShortcuts.ts (hunk 3 above)
Comment (abhinavgautam01) [reply packages/lexical-markdown/src/MarkdownShortcuts.ts]: thanks... folded it into a single ternary: triggerOnEnter || (match && match[0].endsWith(' ')) ? anchorOffset : anchorOffset - 1, which matches the previous branches...

File: packages/lexical-markdown/src/MarkdownTransformers.ts (hunk 4 above)
Comment (abhinavgautam01) [reply packages/lexical-markdown/src/MarkdownTransformers.ts]: agreed that triggerOnEnter: true on CODE was redundant noise. I dropped it and made multiline transformers opt out instead of opt in: registerMarkdownShortcuts now uses byType.multilineElement.filter(t => t.triggerOnEnter !== false) for the Enter path, so the default CODE still gets Enter-at-EOF without an explicit flag and any custom multiline transformer can set triggerOnEnter: false if it should not run on Enter. Updated the MultilineElementTransformer JSDoc to match...

File: packages/lexical-markdown/src/MarkdownShortcuts.ts [hunk 5]
```diff
@@ -437,6 +441,12 @@ export function registerMarkdownShortcuts(
   transformers: Array<Transformer> = TRANSFORMERS,
 ): () => void {
   const byType = transformersByType(transformers);
+  const elementTransformersForEnter = byType.element.filter(
+    t => t.triggerOnEnter,
+  );
+  const multilineElementTransformersForEnter = byType.multilineElement.filter(
+    t => t.triggerOnEnter !== false,
+  );
```
Comment (etrepum) [inline packages/lexical-markdown/src/MarkdownShortcuts.ts]: Why have this filter for multiline transformers at all? It's not tested, and there are no use cases for it in this PR. Just seems like complexity for no reason.

Comment (abhinavgautam01) [reply packages/lexical-markdown/src/MarkdownShortcuts.ts]: you’re right... the filtered list wasn’t buying us anything: we don’t ship an opt-out multiline transformer, it wasn’t covered by tests, and registerMarkdownShortcuts already treats all multiline-element transformers consistently on Enter. i removed the extra variable/filter and now pass byType.multilineElement directly into runMultilineElementTransformers and deleted triggerOnEnter from MultilineElementTransformer (plus Flow) since it only existed for that path...

**Review by etrepum — APPROVED** [review]
